### ISSUE TITLE
charging pile code

### ISSUE THREAD

**Issue by davidzzw** [ChargeTimeEU/Java-OCA-OCPP#226]
Is this code for a single charging station? It does not require a charging pile code, but only needs to know the gun code? As a charging pile operation platform, we need to interface with multiple manufacturers, and differentiated charging piles must have pile codes

**Comment by TVolden** [comment]
Hi @davidzzw,

Thanks for the quest. 

I'm not sure what exactly a charging pile is or a gun code, could you elaborate?

This code is an implementation of the OCA OCPP (charge protocol), for more information, please see: https://www.openchargealliance.org/

To answer you question, I can say that this library can be used to implement software for chargers and operator systems. It may also be applicable for sort of charge box proxy, which basically could act as both a charge box and an operator, if needed.

**Comment by davidzzw** [comment]
> Ah I see. I would guess that such a QR is utilized in a proprietary app of an operator. I would speculate that the unique id in such a QR code refers to a database record for the operator. This would eventually refer the operator to a charge box connection (maintained by a central system to use OCPP terms). Probably used to do a remote start transaction (OCPP request).
> 
> Long story short, there are som abstraction layers between such a QR code and this library.


Yes, it needs customized development, so I couldn't find your project for a long time. Thank you very much
